### ISSUE TITLE
Are PI's documented?

### ISSUE THREAD

**Issue by CxRes** [ietf-tools/xml2rfc#1035]
Are the list of PI's used by xml2rfc documented? Alternatively, is it possible to get a list by looking at the code.

**Comment by jrlevine** [comment]
xml2rfc v2 used PIs but v3 does not. They've been replaced by attributes of the <rfc> element

**Comment by cabo** [comment]
This is not the whole story.

E.g., `<?v3xml2rfc silence=`

It would be good to get an actual list of PIs that have an influence on xml2rfc.

**Comment by jrlevine** [comment]
Looked at the code, there's "silence" and "table_borders" which do the same thing as the corresponding command line switches.
I really wish they hadn't done that since it's yet more stuff that will bite us when people try to render documents a decade from now.

**Comment by kesara** [comment]
Almost complete list of PIs: https://github.com/ietf-tools/xml2rfc/blob/c10a50c427f8b4002d51f020e506dd2c317e615a/xml2rfc/parser.py#L707-L755
But these might get ignored with v3 xml2rfc.

**Comment by jrlevine** [comment]
I think none of them, they'r from v2.

**Comment by cabo** [comment]
This list is the "attribute"-style names for `<?rfc` PIs.
As I mentioned, there are other PIs.
Also, there is no replacement in the v3.rnc for PIs such as "private" or "topblock" (because they don't apply to RFC generation), so these PIs need to stay active.

**Comment by jrlevine** [comment]
I don't understand. Are you still generating v2 XML?

**Comment by rjsparks** [comment]
Yes, kramdown-rfc generates a "hybrid" v2 that pushes the limits of what the v2v3 converter will accept with some v3-like things in it. Its output declares itself to be v2 and it is _designed_ to take advantage of v2v3.

**Comment by cabo** [comment]
How is that issue "completed"?

I have documented what I know in 

https://github.com/cabo/kramdown-rfc/wiki/Syntax#other-xml-features-in-the-yaml-header-other-xml

but I'd sure like to know what else there is.

The table PIs actually address a gap in the vocabulary that urgently needs to be addressed.
It is amazing that we can't do bis documents of v2 RFCs that use tables without seriously damaging readability.

The silence PIs are truly PIs in that they influence the processing (and not even the result), and they are a much better place to store knowledge about which warnings are spurious than the command line.

**Comment by kesara** [comment]
I closed this because in v3 "PIs" are discouraged[^1].
I guess if we are to document this it should got to authors.ietf.org.

[^1]: https://authors.ietf.org/en/upgrading-from-v2

**Comment by cabo** [comment]
There may be some ideology in place that would discourage their use.

This ideology is completely misapplied for the silencing PI.

The table PI indeed should not exist (there should be vocabulary for that), but not documenting it means that we probably never will get around to fixing the feature gap in the actual vocabulary -- thus all but ensuring that the PI will continue to be used, the inverse of what we want to achieve.

So, yes, despite ideology getting in the way, we dutifully need to document these PIs.

**Comment by cabo** [comment]
Thank you!

(And, just to be clear, I'm only talking about the "v3xml2rfc" PIs here, whose name suggests they are specifically intended for the v3 processing rules.)

**Comment by kesara** [comment]
Incomplete list of PIs, this is based on source code (where `xml2rfc` does things differently when these are present: https://notes.ietf.org/xd_32EZ6Qn2ICxHUeqktnQ

**Comment by cabo** [comment]
Thanks!  The problem with this list is that not all of these PIs do something in v3 (e.g., I recently didn't manage to revive a document with topblock/private).
One indicator might be whether we have a test that uses that PI with v3.
(This clearly works for "v3xml2rfc" PIs, but I'm ignorant about the "rfc" ones.)
